src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xunit.Performance;
+using System.Xml;
+using SerializationTypes;
+
+namespace System.Runtime.Serialization
+{
+    #region Performance tests configuration types
+
+    public enum SerializerType
+    {
+        DataContractSerializer,
+        DataContractJsonSerializer,
+        XmlSerializer,
+        JsonNet
+    }
+
+    public enum TestType
+    {
+        SimpleType,
+        ISerializable,
+        Dictionary,
+        ListOfInt,
+        String,
+        ByteArray,
+        XmlElement
+    }
+
+    public interface IPerfTestSerializer
+    {
+        void Init(object obj);
+        void Serialize(object obj, Stream s);
+        void Deserialize(Stream s);
+    }
+
+    public abstract class SerializerFactory
+    {
+        public abstract IPerfTestSerializer GetSerializer(SerializerType serializerType);
+    }
+
+    #endregion
+
+    public class PerformanceTestCommon
+    {
+
+        #region Performance test configuration
+
+        public static IEnumerable<object[]> PerformanceMemberData(SerializerType serializerType)
+        {
+            yield return new object[] { 100, TestType.ByteArray, 1024, serializerType };
+            yield return new object[] { 5, TestType.ByteArray, 1024 * 1024, serializerType };
+            yield return new object[] { 10000, TestType.String, 128, serializerType };
+            yield return new object[] { 10000, TestType.String, 1024, serializerType };
+            yield return new object[] { 1000, TestType.ListOfInt, 128, serializerType };
+            yield return new object[] { 1000, TestType.ListOfInt, 1024, serializerType };
+            yield return new object[] { 1, TestType.ListOfInt, 1024 * 1024, serializerType };
+            yield return new object[] { 1000, TestType.Dictionary, 128, serializerType };
+            yield return new object[] { 100, TestType.Dictionary, 1024, serializerType };
+            yield return new object[] { 10, TestType.SimpleType, 1, serializerType };
+            yield return new object[] { 1, TestType.SimpleType, 15, serializerType };
+            yield return new object[] { 10000, TestType.ISerializable, -1, serializerType };
+            yield return new object[] { 10000, TestType.XmlElement, -1, serializerType };
+        }
+
+        #endregion
+
+        #region Methods to create object for serialization tests
+
+        public static SimpleTypeWihtMoreProperties CreateObject(int height, int parentId, int currentId, int collectionSize, int childListSize)
+        {
+            int index = parentId * childListSize + (currentId + 1);
+            var obj = new SimpleTypeWihtMoreProperties()
+            {
+                IntProperty = index,
+                StringProperty = index + " string value",
+                EnumProperty = (MyEnum)(index % (Enum.GetNames(typeof(MyEnum)).Length)),
+                CollectionProperty = new List<string>(),
```

Comment:
Pass collectionSize to List constructor otherwise the list will reallocate to double it's size every time you add more than it's capacity. Default capacity is 2.